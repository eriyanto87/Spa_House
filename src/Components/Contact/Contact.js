import React, { useState } from "react";
import axios from "axios";
import Nav from "../Nav/Nav";
import Footer from "../Footer/Footer";

export default function Contact() {
  const [serverState, setServerState] = useState({
    submitting: false,
    status: null,
  });
  const handleServerResponse = (ok, msg, form) => {
    setServerState({
      submitting: false,
      status: { ok, msg },
    });
    if (ok) {
      form.reset();
    }
  };
  const handleOnSubmit = (e) => {
    e.preventDefault();
    const form = e.target;
    setServerState({ submitting: true });
    axios({
      method: "post",
      url: "https://formspree.io/f/mvovqrnk",
      data: new FormData(form),
    })
      .then((r) => {
        handleServerResponse(true, "Thanks!", form);
      })
      .catch((r) => {
        handleServerResponse(false, "Please fill out all fields", form);
      });
  };
  return (
    <div className="contact">
      <Nav />
      <section className="contactSection">
        <h3>Contact Us</h3>
        <form className="signup-form" onSubmit={handleOnSubmit}>
          <div>
            <input
              placeholder="First Name"
              type="text"
              name="first-name"
              id="first-name"
            />
          </div>
          <div>
            <input
              type="text"
              name="last-name"
              id="last-name"
              placeholder="Last Name"
            />
          </div>
          <div>
            <input
              placeholder="Email"
              type="text"
              name="username"
              id="username"
              required
            />
          </div>
          <div>
            <textarea
              placeholder="Enter Your Message Here"
              name="message"
              id="message"
            ></textarea>
          </div>
          <button type="submit" disabled={serverState.submitting}>
            SEND
          </button>
          {serverState.status && (
            <p className={!serverState.status.ok ? "errorMsg" : ""}>
              {serverState.status.msg}
            </p>
          )}
        </form>
      </section>
      <Footer />
    </div>
  );
}
