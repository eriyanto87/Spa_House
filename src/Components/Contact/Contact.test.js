import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Contact from "./Contact";

describe("Contact component", () => {
  it("Contact Snapshot", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Contact />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <Contact />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
});
