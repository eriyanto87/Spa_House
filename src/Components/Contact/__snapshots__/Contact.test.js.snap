// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component Contact Snapshot 1`] = `
<div
  className="contact"
>
  <div
    className="nav"
  >
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home 
        </a>
      </li>
      <li>
        <a
          href="/about"
          onClick={[Function]}
        >
          About 
        </a>
      </li>
      <li>
        <a
          href="/treatments"
          onClick={[Function]}
        >
          Treatments 
        </a>
      </li>
      <li>
        <a
          href="/pricing"
          onClick={[Function]}
        >
          Pricing 
        </a>
      </li>
      <li>
        <a
          href="/contact"
          onClick={[Function]}
        >
          Contact Us 
        </a>
      </li>
      <li>
        <a
          href="/gallery"
          onClick={[Function]}
        >
          Gallery
        </a>
      </li>
    </ul>
  </div>
  <section
    className="contactSection"
  >
    <h3>
      Contact Us
    </h3>
    <form
      className="signup-form"
      onSubmit={[Function]}
    >
      <div>
        <input
          id="first-name"
          name="first-name"
          placeholder="First Name"
          type="text"
        />
      </div>
      <div>
        <input
          id="last-name"
          name="last-name"
          placeholder="Last Name"
          type="text"
        />
      </div>
      <div>
        <input
          id="username"
          name="username"
          placeholder="Email"
          type="text"
        />
      </div>
      <div>
        <textarea
          id="message"
          name="message"
          placeholder="Enter Your Message Here"
          type="message"
        />
      </div>
      <button
        disabled={false}
        type="submit"
      >
        SEND
      </button>
    </form>
  </section>
  <div>
    <p>
      Â© 2020
    </p>
  </div>
</div>
`;
